### ISSUE TITLE
[Bug]: Crash with items expanding from tags

### ISSUE THREAD

**Issue by ejektaflex** [ejektaflex/Bountiful#234]
### What happened?

I recently received a report containing a ticking entity error:
```txt
net.minecraft.class_151: Non [a-z0-9_.-] character in namespace of location: #croptopia:nuts
...
Name: bountiful:board-be // io.ejekta.bountiful.content.board.BoardBlockEntity
```
It seems like my tag items are not expanding into regular items when being selected, or their validity check is failing. I'll look into this further next week (I am on vacation), but for now the easy fix is to place this file in `config/bountiful/bounty_pools/farmer_objs.json`:
```json
{
	"content": {
		"farmer_obj_nuts": null,
		"farmer_obj_peppers": null,
		"farmer_obj_sauces": null,
		"farmer_obj_flourables": null
	}
}
```

### Mod Version

1.20.1

### Mod Loader

Fabric

### Mod Loader Version

N/A

**Comment by YuRaNnNzZZ** [comment]
Not just Croptopia.
Forge 47.0.35 (1.20.1), mod version 6.0.1
[crash-2023-06-29_14.28.37-server.txt](https://github.com/ejektaflex/Bountiful/files/11905165/crash-2023-06-29_14.28.37-server.txt)

I think all "item" type tag entries need to be switched to "item_tag" ones
https://github.com/ejektaflex/Bountiful/blob/ca3324a6a646cce8e571082936cf8630f54cb93e/datagen/data/common/bountiful/bounty_pools/shepherd_objs.json#L38-L39
https://github.com/ejektaflex/Bountiful/blob/ca3324a6a646cce8e571082936cf8630f54cb93e/datagen/data/common/bountiful/bounty_pools/shepherd_objs.json#L49-L50

**Comment by ejektaflex** [comment]
~~Correct!~~ Not correct, after looking at it a bit closer (though I can't look at the log while I'm away from home). Item type objectives starting with # are different from item_tag bounties.

`item` objectives starting with #: will turn into a random item from that tag (`#minecraft:axes` will turn into a random axe objective)

Whereas, `item_tag` objectives will ask for any item from said tag (`minecraft:axes` will be an objective of "get X number of any axes").

As such, it's not a problem of bounty type, it's more a problem that the hashtag on item objectives seems to not be parsed correctly.

**Comment by Shadowpudding** [comment]
I just got the wool crash as well. I am just going to disable the mod till this gets fixed or I could test around a bit and find out what other objects cause the game to crash

**Comment by Shadowpudding** [comment]
I played around for a bit with just bountiful and no other mod. I only got the wool crash. Because it will continue to crash i am unable to find out with degree the wool is a part of, but it seems that only that one is affected.With my main modpack i opened a testing world that crashed so bad that the world got corrupted

**Comment by OBS3SS1ON** [comment]
Same error 1.20.1-forge-47.0.19
https://pastebin.com/Re0dyzR9

The game crashed whilst ticking block entity
Error: net.minecraft.ResourceLocationException: Non [a-z0-9_.-] character in namespace of location: #minecraft:wool

**Comment by Xicad** [comment]
Same error 1.20.1, fabric 0.14.21 mod version 6.0.1

https://pastebin.com/nF9ck8uV

**Comment by ejektaflex** [comment]
Yes, this will happen with any item objectives that expands via tag, most notable with Croptopia. As said before, I'll get around to fixing it next week :) in the meantime, you can set those objectives to null via a config file matching the pool name and it should act as a temporary fix.

**Comment by GuyInUniverse** [comment]
I'm getting the same crash every once in a while in 1.20.1 on the latest Fabric mod loader. Sometimes I have to remove the mod in order to load my world again. Aside from performance mods, the only additional mods I had were Terralith, ChoiceTheorem's Overhauled Villages and Comforts.


Description: Ticking block entity
net.minecraft.class_151: Non [a-z0-9_.-] character in namespace of location: #minecraft:wool

Details:
	Name: bountiful:board-be // io.ejekta.bountiful.content.board.BoardBlockEntity


Edit: Also thanks for the great mod, I've been really enjoying it!

**Comment by ejektaflex** [comment]
Yep! I know the fix, I'd just been busy. Look forward to a hotfix in the next day or two. Thank you!

**Comment by ejektaflex** [comment]
For people without Croptopia wanting a quick "vanilla" fix (@GuyInUniverse, @Xicad, @OBS3SS1ON, @Shadowpudding), you can put also this in `config/bountiful/bounty_pools/shepherd_objs.json`:
```json
{
	"content": {
		"shepherd_obj_wool_random": null,
		"shepherd_obj_banner_random": null
	}
}
```
And this in `config/bountiful/bounty_pools/shepherd_rews.json`:
```json
{
	"content": {
		"shepherd_rew_carpet_random": null,
		"shepherd_rew_bed_random": null,
 		"shepherd_rew_banner_random": null
	}
}
```

**Comment by GuyInUniverse** [comment]
@ejektaflex Thank you!

**Comment by ejektaflex** [comment]
This has been fixed in 6.0.2 and released on both CurseForge and Modrinth. If any of you applied these changed to the configs, you may remove them after updating. Thanks much!
